docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,242 @@
+    "DB_UNIT= <katib-db/0 | kfp-db/0>\n",
+    "\n",
+    "# obtain username and password of existing MariadDB database from DB relation\n",
+    "# in Charmed Kubeflow 1.7 username used is `root` and password is specified in `mysql` relation by\n",
+    "# 'root_password'\n",
+    "DB_MYSQL_RELATION_ID=$(juju show-unit $DB_UNIT | yq '.[] | .relation-info | select(.[].endpoint == \"mysql\") | .[0] | .relation-id')\n",
+    "DB_USER=root\n",
+    "DB_PASSWORD=$(bash -c \"juju run --unit $DB_UNIT 'relation-get -r $DB_MYSQL_RELATION_ID - $DB_UNIT' | grep root_password\" | awk '{print $2}')\n",
+    "DB_IP=$(juju show-unit $DB_UNIT/0 | yq '.[] | .address')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy MySQL databases\n",
+    "\n",
+    "Deploy new MySQL database and obtain credentials by executing the following commands:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# replace MYSQL_DB with desired application name: `katib-db-mysql` or `kfp-db-mysql`\n",
+    "MYSQL_DB= <katib-db-mysql/0 | kfp-db-mysql/0>\n",
+    "\n",
+    "# deploy new MySQL database charm\n",
+    "juju deploy mysql-k8s $MYSQL_DB --channel 8.0/stable --trust\n",
+    "\n",
+    "# obtain username and password of new MySQL database from MySQL charm\n",
+    "MYSQL_DB_USER=$(juju run-action $MYSQL_DB get-password --wait | yq '.[] | .results.username')\n",
+    "MYSQL_DB_PASSWORD=$(juju run-action $MYSQL_DB get-password --wait | yq '.[] | .results.password')\n",
+    "MYSQL_DB_IP=$(juju show-unit $MYSQL_DB | yq '.[] | .address')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Katib DB migration\n",
+    "\n",
+    "Use credentials and information obtained in previous steps for `katib-db` to perform database migration by executing the following commands:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# ensure that there are no new connections are made and that dababase is not altered\n",
+    "# remove relation between katib-db-manager charm and katib-db charm\n",
+    "juju remove-relation katib-db-manager katib-db\n",
+    "\n",
+    "# connect to unit's IP address using mysql client and verify that `katib` database exists\n",
+    "# this command will log you into mysql monitor prompt\n",
+    "mysql --host=$DB_IP --user=$DB_USER --password=$DB_PASSWORD\n",
+    "\n",
+    "# exit mysql client session\n",
+    "\n",
+    "# create backup of all databases file using `mysqldump` utility, username, password, and unit's IP address\n",
+    "# onbtained earlier\n",
+    "mysqldump --host=$DB_IP --user=$DB_USER --password=$DB_PASSWORD --column-statistics=0 --databases katib  > katib-db.sql\n",
```

Comment:
So the backup is a `.sql` file, worth mentioning that with this command a file will be generated and saved in the CWD.

Reply:
Will do.